### PR TITLE
docs(aio): Updated usage of Observables and redirects in router guide.

### LINKED ISSUE #16377: docs: router - improve docs on the router redirect
Need to be documented:
- for absolute redirect, the router use the query parameters and the fragment from the `redirectTo` config,
- for relative redirect, the router use the query params and the fragment from the URL from which we are redirected.

Also it is possible to use positional parameters and query params in the redirected url:
- `{path: 'a/:aid/b/:bid', redirectTo: 'newa/:aid/newb/:bid'}`
- `{path: 'b/:id', redirectTo: '/absolute/:id?a=1&b=:b#f1'}`

See apply_redirect_spec.ts for details

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

cc: @wardbell 

Closes #16377 #17901

### CODE REVIEW

File: aio/content/examples/router/src/app/crisis-center/crisis-list.component.1.ts [hunk 1]
```diff
@@ -10,35 +10,34 @@ import { Observable }            from 'rxjs/Observable';
...
-    </ul>`
+    </ul>
+
+    <router-outlet></router-outlet>
+    `
   // #enddocregion relative-navigation-router-link
 })
 export class CrisisListComponent implements OnInit {
-  crises: Observable<Crisis[]>;
+  crises$: Observable<Crisis[]>;
```
Comment (petebacondarwin) [inline aio/content/examples/router/src/app/crisis-center/crisis-list.component.1.ts]: So we are going $-postfix for observables now?
Is this consistent across the guide or a slip?

File: aio/content/examples/router/src/app/crisis-center/crisis.service.ts [hunk 2]
```diff
@@ -12,28 +16,27 @@ const CRISES = [
...
 
-let crisesPromise = Promise.resolve(CRISES);
-
 import { Injectable } from '@angular/core';
 
 @Injectable()
 export class CrisisService {
-
   static nextCrisisId = 100;
+  crises: BehaviorSubject<Crisis[]> = new BehaviorSubject<Crisis[]>(CRISES);
```
Comment (petebacondarwin) [inline aio/content/examples/router/src/app/crisis-center/crisis.service.ts]: What no `$` postfix?

File: aio/content/guide/router.md [hunk 3]
```diff
@@ -1813,20 +1767,19 @@ pull the hero `id` from the parameters and retrieve the hero to display.
...
 You might think to use the RxJS `map` operator.
 But the `HeroService` returns an `Observable<Hero>`.
-Your subscription wants the `Hero`, not an `Observable<Hero>`.
 So you flatten the `Observable` with the `switchMap` operator instead.
 
 The `switchMap` operator also cancels previous in-flight requests. If the user re-navigates to this route
 with a new `id` while the `HeroService` is still retrieving the old `id`, `switchMap` discards that old request and returns the hero for the new `id`.
 
-Finally, you activate the observable with `subscribe` method and (re)set the component's `hero` property with the retrieved hero.
+The observable `Subscription` will be handled by the `AsyncPipe` and the component's `hero` property will be re(set) with the retrieved hero.
```
Comment (petebacondarwin) [inline aio/content/guide/router.md]: re(set) or (re)set?

File: aio/content/guide/router.md [hunk 4]
```diff
@@ -3541,8 +3459,12 @@ Be explicit. Implement the `Resolve` interface with a type of `Crisis`.
 Inject the `CrisisService` and `Router` and implement the `resolve()` method.
 That method could return a `Promise`, an `Observable`, or a synchronous return value.
 
-The `CrisisService.getCrisis` method returns a promise.
-Return that promise to prevent the route from loading until the data is fetched.
+The `CrisisService.getCrisis` method returns an Observable.
+Return that observable to prevent the route from loading until the data is fetched.
+The `Router` guards require an Observable to `complete`, meaning it has emitted all
+of its values. You use the `take` operator with an argument of `1` to ensure that the
+Observable completes after retreiving the first value from the Observable returned by the
```
Comment (petebacondarwin) [inline aio/content/guide/router.md]: retrieving - typo

File: aio/content/guide/router.md [hunk 5]
```diff
@@ -4041,6 +3966,52 @@ Verify this by logging in to the `Admin` feature area and noting that the `crisi
...
+</code-example>
+
+
+You'll notice two different types of redirects. The first change is from  `/heroes` to `/superheroes` without any parameters. This is a straightforward redirect, unlike the change from `/hero/:id` to `/superhero/:id`, which includes the `:id` route parameter. Router redirects also use powerful pattern matching, so the `Router` inspects the URL and replaces route parameters in the `path` with their appropriate destination. Previously, you navigated to a URL such as `/hero/15` with a route parameter `id` of `15`.
+
+<div class="l-sub-section">
+
+The `Router` also supports [query parameters](#query-parameters) and the [fragment](#fragment) when using redirects.
+
+* When using absolute redirect, the `Router` will use the query parameters and the fragment from the redirectTo in the route config.
```
Comment (petebacondarwin) [inline aio/content/guide/router.md]: This phrase:

> When using absolute redirect,

doesn't quite read right. But I am not sure what would be better.

**Review by petebacondarwin — APPROVED** [review]
A couple of minor typos and comments but LGTM.

File: aio/content/guide/router.md (hunk 5 above)
Comment (brandonroberts) [inline aio/content/guide/router.md]: Fixed

File: aio/content/guide/router.md (hunk 4 above)
Comment (brandonroberts) [inline aio/content/guide/router.md]: Fixed

File: aio/content/guide/router.md (hunk 3 above)
Comment (brandonroberts) [inline aio/content/guide/router.md]: Fixed with the latter

File: aio/content/examples/router/src/app/crisis-center/crisis.service.ts (hunk 2 above)
Comment (brandonroberts) [inline aio/content/examples/router/src/app/crisis-center/crisis.service.ts]: Fixed

File: aio/content/examples/router/src/app/crisis-center/crisis-list.component.1.ts (hunk 1 above)
Comment (brandonroberts) [inline aio/content/examples/router/src/app/crisis-center/crisis-list.component.1.ts]: I think its a common convention at this point

**Comment by brandonroberts** [comment]
@petebacondarwin minor changes and typos are complete. Thanks!

**Comment by brandonroberts** [comment]
@wardbell @petebacondarwin can we land this?

**Comment by petebacondarwin** [comment]
Can you rebase it @brandonroberts ? 
It is currently failing the AIO build but these have changed since the last time Travis ran.

**Comment by brandonroberts** [comment]
@petebacondarwin done

**Comment by petebacondarwin** [comment]
@brandonroberts the router e2e tests failed: https://travis-ci.org/angular/angular/jobs/271225109#L1975

**Comment by brandonroberts** [comment]
@petebacondarwin e2e tests are fixed

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
@brandonroberts - great! Should this be backported to the patch release version too?

**Comment by brandonroberts** [comment]
@petebacondarwin I don't see why not. Do I need to open a new PR for that?

**Comment by petebacondarwin** [comment]
Nope! Just helps me decide what labels to attach :-)

**Comment by brandonroberts** [comment]
:-)

**Pull request closed without merging**